### DISCUSSION TITLE
Rule for auto power off with timer reset if consumption exceeds value for a specific period

### DISCUSSION THREAD

**Discussion by laterne-rouge** [arendst/Tasmota#19055] [Q&A] [answered]
Hey there,
I am new to Tasmota and currently trying to program a rule for following scenario.

I have a plug with power metering. For my espresso machine, I would like to implement an automatic power off function. After powering on the machine, a timer for 45 minutes should be started after which the plug turns the power off. 

If the machine uses more than 1000 watts for more than 10 seconds (an espresso is brewed) it should reset the timer to 45 minutes.

I tried it with following code, but it does not work: 

`Rule1 ON Power on DO rule2 1; ruletimer1 600 ENDON

Rule2 ON Energy#Power>1000 DO ruletimer2 10 ENDON ON rules#timer2=1 DO ruletimer1 600 ENDON

Rule3  ON Rule#timer1=1 DO Power off ENDON

Rule1 1 
Rule1 5 
Rule2 5
Rule3 1 `

Any help or suggestions would be highly appreciated.

Thanks.

**Comment by sfromis** [comment]
`rules#timer2=1` --> `rules#timer=2`

**Comment by pkkrusty** [comment]
Also need backlog in rule1. 

Also 45 minutes = 2700 seconds. Not sure why you have 600 seconds. 

Currently, the wattage will trigger the ten second timer as soon as it goes above 1000W.  But you want the rule to be triggered only after the power has been >1000W for more than 10 seconds, right? Is there any scenario where the wattage would go above 1000W for less than 10 seconds?

45 minutes seems like a long time. Any reason for keeping it on that long? Why not turn it off 2 minutes after the wattage returns to baseline?

**Reply by laterne-rouge** [reply]
Thanks for your responses. With both changes (backlog and rules#timer) the power off function works once the first timer is reached.

There are scenarios where the wattage goes above 1000W for less than ten seconds, in which case the first timer should not be reset. Not quite sure how I would program this. Any suggestions?

**Reply by sfromis** [reply]
You can use another RuleTimer to delay any action for "above 1000W" till you know you want it

**Comment by pkkrusty** [comment, marked as answer]
```
Rule1 
on Energy#Power>1000 DO backlog ruletimer2 10 endon
on Energy#Power<=1000 do backlog ruletimer2 0 endon

Rule2 
on Power1#state=1 DO ruletimer1 2700 endon
on rules#timer2=1 DO ruletimer1 2700 endon
on rules#timer=1 do power1 0 endon

Rule1 1
Rule2 1
Rule1 5
```

I think this will get you what you want. Rule1 is one-shot, so the ten second timer runs until it gets shut off by the power falling below 1000w (the trigger condition changing).  If it reaches completion, it triggers a restart of the 45 minute timer.

**Reply by sfromis** [reply]
Nitpicking....
`on rules#timer2=1 DO ruletimer1 2700 endon`
should more probably be
`on rules#timer=2 DO ruletimer1 2700 endon`

**Reply by pkkrusty** [reply]
Ah yes, you're right.
